contracts/SSVNetwork.sol

```diff
@@ -320,20 +321,23 @@ contract SSVNetwork is UUPSUpgradeable, Ownable2StepUpgradeable, ISSVNetwork {
             _deposit(amount);
         }
 
-        emit ValidatorAdded(msg.sender, operatorIds, publicKey, shares, cluster);
+        emit ValidatorAdded(msg.sender, operatorIds, publicKey, signatureShares, cluster);
     }
 
     function removeValidator(
         bytes calldata publicKey,
-        uint64[] memory operatorIds,
+        uint64[] calldata operatorIds,
         Cluster memory cluster
     ) external override {
-        bytes32 hashedValidator = keccak256(publicKey);
-        address validatorOwner = validatorPKs[hashedValidator].owner;
-        if (validatorOwner == address(0)) {
+        bytes32 hashedValidator = keccak256(abi.encodePacked(publicKey, msg.sender));
+        Validator memory validator = validatorPKs[hashedValidator];
+
+        if (!validator.active) {
```

Comment:
Because active will mean something else in the future lets think on how to check this without having to use this flag